frontend-react/src/pages/admin/AdminMessageTestingPage/AdminMessageTestingPage.tsx

```diff
@@ -0,0 +1,132 @@
+    StepTwo = "MessageTestResults",
+}
+
+const AdminMessageTestingPage = () => {
+    const { orgname, receivername } = useParams<EditReceiverSettingsParams>();
+    const crumbProps: CrumbsProps = {
+        crumbList: [
+            {
+                label: FeatureName.RECEIVER_SETTINGS,
+                path: `/admin/orgreceiversettings/org/${orgname}/receiver/${receivername}/action/edit`,
+            },
+            { label: FeatureName.MESSAGE_TESTING },
+        ],
+    };
+
+    // Sets step of the Message Testing process laid out in the MessageTestingSteps enum
+    const [currentMessageTestStep, setCurrentMessageTestStep] = useState<MessageTestingSteps>(
+        MessageTestingSteps.StepOne,
+    );
+
+    // Sets data required for the MessageTestingForm
+    const { data: messageData, isDisabled } = useTestMessages();
+    const { setRequestBody, isLoading, data: testResultData } = useTestMessageResult();
+    const [selectedOption, setSelectedOption] = useState<RSMessage | null>(null);
+    const [currentTestMessages, setCurrentTestMessages] = useState<RSMessage[]>(messageData);
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        setRequestBody(selectedOption?.reportBody ? selectedOption.reportBody : null);
+        setCurrentMessageTestStep(MessageTestingSteps.StepTwo);
+    };
+
+    // If a user is _not_ a PrimeAdmin, block this feature
+    if (isDisabled) {
+        return (
+            <StaticAlert
+                type={StaticAlertType.Error}
+                heading="Access denied"
+                message="Please try again as a PrimeAdmin"
+            />
+        );
+    }
+
+    return (
+        <>
+            <Helmet>
+                <title>Message testing - ReportStream</title>
+            </Helmet>
+            <GridContainer>
+                <Crumbs {...crumbProps}></Crumbs>
+            </GridContainer>
+            <AdminFormWrapper
+                header={
+                    <>
+                        <Title title={"Message testing"} />
+                        <h2 className="margin-bottom-0">
+                            <span className="text-normal font-body-md text-base margin-bottom-0">
+                                Org name: {orgname}
+                                <br />
+                                Receiver name: {receivername}
+                            </span>
+                        </h2>
+                    </>
+                }
+            >
+                <GridContainer>
+                    <p>
+                        Test a message from the message bank or by entering a custom message. You can view test results
+                        in this window while you are logged in. To save for later reference, you can open messages, test
+                        results and output messages in separate tabs.
+                    </p>
+                    <hr />
+                    {isLoading ? (
```

Comment:
The loading condition will never be true under suspense, as the component will not be rendered until the data is fetched.

Reply:
Correct, this was written alongside another note about using useSuspenseQuery that is no longer relevant